Quake/sys_sdl_unix.c

```diff
@@ -321,6 +321,21 @@ static void Sys_GetBasedir (char *argv0, char *dst, size_t dstsize)
 {
 	char	*tmp;
 
+	#ifdef PLATFORM_HAIKU
+	if (realpath(argv0, dst) == NULL)
+	{
+		perror("realpath");
+		if (getcwd(dst, dstsize - 1) == NULL)
+	_fail:		Sys_Error ("Couldn't determine current directory");
+	}
+	else
+	{
+		/* strip off the binary name */
+		if (! (tmp = strdup (dst))) goto _fail;
+		q_strlcpy (dst, dirname(tmp), dstsize);
+		free (tmp);
```

Comment:
How are you stripping off the binary name here??

Reply:
To be completely honest with you, I just copied that code from the OSX version of Sys_GetBasedir() at line 309 in that file. I don't know why it works.

Reply:
Agh.. Didn't notice that you're passing `dirname(tmp)` to strlcpy. Sorry for the noise.

Will pull this in, soon